contracts/smart-contract-wallet/modules/SessionKeyModule.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {BaseAuthorizationModule, UserOperation, ISignatureValidator} from "./BaseAuthorizationModule.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@account-abstraction/contracts/core/Helpers.sol";
+import "./SessionValidationModules/ISessionValidationModule.sol";
+
+struct SessionStorage {
+    bytes32 merkleRoot;
+}
+
+contract SessionKeyManager is BaseAuthorizationModule {
+    /*
+     * @dev mapping of Smart Account to a SessionStorage
+     * Info about session keys is stored as root of the merkle tree built over the session keys
+     */
+    mapping(address => SessionStorage) internal userSessions;
+
+    function getSessionKeys(
+        address smartAccount
+    ) external view returns (SessionStorage memory) {
+        return userSessions[smartAccount];
+    }
+
+    function setMerkleRoot(bytes32 _merkleRoot) external {
+        _setSessionData(msg.sender, _merkleRoot);
```

Comment:
so we will make SmartAccount call this method? 
what would the transaction flow look like from EntryPoint?

Reply:
EP=>SA.executeCall=>SKManager.setMerkleRoot
it's illustrated here 
https://github.com/bcnmy/scw-contracts/blob/19d046210f70557ccb31f8564658778b63890e42/test/module/SessionKeyManager.Module.specs.ts#L98-L132

Reply:
got it, resolved.